core/engine_test.go

```diff
@@ -915,3 +915,49 @@ func TestEmittedMetricsWhenScalingDown(t *testing.T) {
 	durationSum := getMetricSum(collector, metrics.IterationDuration.Name)
 	assert.InDelta(t, 1.7, durationSum/(1000*durationCount), 0.1)
 }
+
+func TestMinIterationDuration(t *testing.T) {
+	t.Parallel()
+
+	runner, err := js.New(
+		&lib.SourceData{Filename: "/script.js", Data: []byte(`
+		import { sleep } from "k6";
+
+		export let options = {
+			minIterationDuration: "1s",
+			vus: 2,
+			vusMax: 2,
+			duration: "1.5s",
+		};
+
+		export default function () {
+			sleep(0.1);
```

Comment:
Not sure if this won't be too fragile especially with a lot of parallel tests. 

Maybe something like a single metric that gets added to in the default. A test duration of 1.9s so we will always add twice to the metric but  will probably be less fragile than sleep that will sleep for *at least* 0.1s.